src/API/Models/DesignerModel.ts

```diff
@@ -82,28 +105,39 @@ export const readAllDesigners = async (filters: DesignerFilters) => {
       },
       baseAccount: {
         select: {
-          avatarUrl: true
+          avatarUrl: true,
+          gender: true,
+          firstName: true,
+          lastName: true
         }
       }
     }
   });
 
-
-  return designers.map(designer => {
+  const filteredDesigners = designers.map(designer => {
     const workingDays: WorkingHours = JSON.parse(designer.workingDays as unknown as string);
     const { open, openUntil } = isOpenNow(workingDays);
 
     return {
       baseAccountId: designer.baseAccountId,
       ordersFinished: designer.ordersFinished,
-      location: designer.location, // Include location instead of address
+      location: designer.location,
       yearsExperience: designer.yearsExperience,
       rating: designer.reviews.length ? designer.reviews.reduce((sum: number, review: { rating: number }) => sum + review.rating, 0) / designer.reviews.length : 0,
       avatarUrl: designer.baseAccount.avatarUrl,
+      gender: designer.baseAccount.gender,
+      name: `${designer.baseAccount.firstName} ${designer.baseAccount.lastName}`,
       openNow: open,
       openUntil: open ? openUntil : null
     };
   });
+
+  const openFilteredDesigners = openNow !== undefined
+    ? filteredDesigners.filter(designer => designer.openNow === openNow)
+    : filteredDesigners;
+
+  const paginatedDesigners = openFilteredDesigners.slice((page - 1) * limit, page * limit);
```

Comment:
This is not the most efficient way to fetch data and apply pagination since you fetch EVERYTHING then filter it.
Sounds like a really high space & time complexity solution.
but again if it works then it's good.

Reply:
<img width="201" alt="image" src="https://github.com/MIS-Team24/khoyout-backend/assets/73319030/c8f381e2-c466-432f-a32f-d5f7d3addf2d">